src/Tools/ExternalAccess/Razor/DebuggerFindReferencesService.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.FindUsages;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Internal.Log;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    [Export]
+    [Shared]
+    internal sealed class DebuggerFindReferencesService
+    {
+        private readonly IThreadingContext _threadingContext;
+        private readonly IEnumerable<Lazy<IStreamingFindUsagesPresenter>> _streamingPresenters;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public DebuggerFindReferencesService(
+            IThreadingContext threadingContext,
+            [ImportMany] IEnumerable<Lazy<IStreamingFindUsagesPresenter>> streamingPresenters)
+        {
+            _threadingContext = threadingContext;
+            _streamingPresenters = streamingPresenters;
+        }
+
+        public async Task FindSymbolReferencesAsync(ISymbol symbol, Project project, CancellationToken cancellationToken)
+        {
+            var streamingPresenter = GetStreamingPresenter();
+
+            // Let the presenter know we're starting a search.  It will give us back
+            // the context object that the FAR service will push results into.
+            var context = streamingPresenter.StartSearch(EditorFeaturesResources.Find_References, supportsReferences: true);
+
+            using (Logger.LogBlock(
+                FunctionId.CommandHandler_FindAllReference,
```

Comment:
I also don't think we want this since we have this to track our find all references command usages. if debugger team want to track their usage of find all reference, they should report telemetry themselves.

Reply:
it would be nice if there is just XXX.FARWithUI(supportsReferences) and OB and other just use that. we could add internal one TryFindAllReferenceAsync in VSWorkspace and all just call that as well?

Reply:
> I also don't think we want this since we have this to track our find all references command usages. if debugger team want to track their usage of find all reference, they should report telemetry themselves.

Makes sense. I removed our own telemetry from this API.

@mpeyrotc if you want telemetry for these operations you should wrap your call in VsDebugPresentationPackage to track it.

> it would be nice if there is just XXX.FARWithUI(supportsReferences) and OB and other just use that. we could add internal one TryFindAllReferenceAsync in VSWorkspace and all just call that as well?

Right now this is the only caller that I'm aware of. Do we have indications that other external components need _this specific_ functionality?

Reply:
📝 This is the kind of refactoring that we can provide later without breaking external users. This assembly will always need to have an API with this signature, even if we relocate the implementation:

```csharp
public async Task FindSymbolReferencesAsync(ISymbol symbol, Project project, CancellationToken cancellationToken)
```